#!/usr/bin/env python
PACKAGE = "image_cloud"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("subscribe_topic_pcl",		str_t,	 	0, "subscribe_topic_pcl", "")
gen.add("subscribe_topic_img_info",	str_t,	 	0, "subscribe_topic_img_info", "")
gen.add("publish_topic",			str_t,		0, "publish_image topic", "" )
gen.add("image_tf_frame_id",		str_t,	 	0, "image tf frame id", "")
gen.add("use_reference", 			bool_t, 	0, "use reference frame for time transformation and publishing", False)
gen.add("reference_frame",			str_t,	 	0, "reference frame for transformation and pointcloud publishing", "")
gen.add("point_size", 				int_t, 		0, "draw point size", 1, 1, 30)
gen.add("resize_faktor_x",			double_t,	0, "resize_faktor_x", 1, 0.01, 4 )
gen.add("resize_faktor_y",			double_t,	0, "resize_faktor_y", 1, 0.01, 4 )
gen.add("queue_size",				int_t,		0, "message filter queue", 30, 1, 100 )
gen.add("normal_search_radius",		double_t,	0, "normal search radious for normal features", 0.2, 0.001, 2 )

feature_enum = gen.enum([ 	gen.const("Intesity",    	int_t, 0, "Intensity"),
                       		gen.const("Normal",     		int_t, 1, "Normal"),
                       		gen.const("Normal_and_Intensity",   int_t, 2, "Normal and Intensity")],
                     		"select filter")

gen.add("feature", int_t, 0, "filter selector", 0, 0, 2, edit_method=feature_enum)

exit(gen.generate(PACKAGE, "image_cloud", "pcl_image"))